pkg/reconciler/knativeserving/knativeserving_controller.go

```diff
@@ -290,5 +290,13 @@ func (r *Reconciler) deleteObsoleteResources(manifest *mf.Manifest, instance *se
 	if err := manifest.Client.Delete(resource); err != nil {
 		return err
 	}
+	// Remove Service activator from 0.12, since the new service is named activator-service
```

Comment:
can we add an e2e test for this to verify that it is actually removed?

Reply:
do we have a test that detected there were leftover objects?

Reply:
Upgrade tests may be able to make sure these resources are removed.

Reply:
can we update the upgrade test to ensure that after upgrading, there are no objects labeled for the previous version?

Adding logic that doesnt have corresponding tests makes me wonder if it is actually working or not.

Most objects have a release label, would it be safe to delete all objects from the old release?

Reply:
It's harmless for this to be in here, since it'll fail to delete anything silently, but it appears the name didn't change: https://github.com/knative/serving-operator/blob/326cbf7991bc801e631b2cef729be6ad7ca51991/cmd/manager/kodata/knative-serving/0.12.1.yaml#L784

Reply:
@jcrossley3 @k4leung4 Was it just an illusion, or something wrong with my eyes?
I caught a Service named "activator", but cannot find it any more.